posts/frangeris/06.stale-cache.md

```diff
@@ -0,0 +1,136 @@
+---
+title: "Stale cache, the holy grail of performance"
+published: true
+description: The perfect cache system for an API service
+tags: "cache, cdn, api, performance"
+cover_image: ./assets/06.cover.png
+---
+
+Cache on APIs is a must-have feature for any service that wants to scale. It's a way to reduce the load on your servers and improve the response time for your users by serving the same content multiple times, getting the same result without having to process the request again.
+
+The issue is that... implementing cache systems is hard, tricky and can cause a lot of headaches when it comes to invalidating the cache, purging and keeping the cache fresh with the latest data.
+
+In this article, we will explore the concept of stale cache, test by examples and see how CDNs providers like Bunny CDN can help us to keep the cache fresh without having to purge it every time a new request is made.
+
+---
+
+Let's start by breaking down the pieces.
+
+## ⌛️ Response time
+
+When you make a request to an API, the response time is the time it takes for the server to process the request and send back the response. This time can vary depending on the server's load, the network latency, and the complexity of the request.
+
+Typically the composition of this time is affected multiple factors like:
+
+- **Network latency**: The time it takes for the request to travel from the client to the server.
+- **DNS lookup time**: The time it takes for the client to resolve the server's domain name to an IP address.
+- **Connections and SSL Handshakes**: Set of ticks and checks that the server needs to do before sending the response.
+
+And when the requests finally arrive, then in the server side we have:
+
+- **Processing time**: Time it takes for the server to process the request and send back the response.
+- **Response size**: The size of the response data, which affects the time it takes to transfer the data over the network.
```

Comment:
```suggestion
- **Response size**: this affects the time it takes to transfer the data over the network.
```

It seems unnecessary to repeat what is obvious from the title. Maybe some other definition before talking about what it does, but not the same words the other way around.